src/main/java/lotto/domain/TicketOffice.java

```diff
@@ -0,0 +1,39 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class TicketOffice {
+    private final LottoTicketPrice lottoTicketPrice;
+
+    public TicketOffice(LottoTicketPrice lottoTicketPrice) {
+        this.lottoTicketPrice = lottoTicketPrice;
+    }
+
+    public LottoTickets sale(NumberOfTicket numberOfTicket) {
+        LottoTickets lottoTickets = new LottoTickets(new ArrayList<>());
+
+        for (int i = 0; i < numberOfTicket.count(); i++) {
+            lottoTickets.add(new LottoTicket(new LottoNumbers()));
+        }
+
+        return lottoTickets;
+    }
+
+    public NumberOfTicket numberOfTicket(PurchaseAmount purchaseAmount) {
+        return new NumberOfTicket(purchaseAmount.purchaseAmount() / lottoTicketPrice.price());
```

Comment:
```suggestion
        return new NumberOfTicket(purchaseAmount, lottoTicketPrice);
```
`NumberOfTicket` 을 단순히 티켓수를 가진 객체가 아니라 `PurchaseAmount` 객체와 `LottoTicketPrice` 객체를 통해 개산해 낼 수 있는 능력이 있는 객체로 업그레이드 해 봅시다.  객체지향의 장점은 이렇게 위임함으로서 유연성이 높아지는데 있어요. 예를들어 할인같은 계산방식이 변경된다면  `TicketOffice` 는 영향받지 않고 `NumberOfTicket` 이 책임질 수 있을것 같아요.

물론 실무에선 이렇게 단순하진 않겠지만 객체지향적 사고, 훈련을 하는 차원에선 의미가 있을것 같습니다. 😀